worlds/ff4fe/__init__.py

```diff
@@ -0,0 +1,397 @@
+
+    def create_event(self, event: str) -> FF4FEItem:
+        # while we are at it, we can also add a helper to create events
+        return FF4FEItem(event, ItemClassification.progression, None, self.player)
+
+    def create_items(self) -> None:
+        exclude = [item for item in self.multiworld.precollected_items[self.player] if item != "None"]
+        item_pool_result = create_itempool(locations.all_locations, self)
+        item_pool = item_pool_result[0]
+        self.chosen_character = item_pool_result[1]
+        chosen_character_placed = False
+        for item in map(self.create_item, item_pool):
+            if item.name == self.chosen_character and not chosen_character_placed:
+                (self.multiworld.get_location("Starting Character 1", self.player)
+                 .place_locked_item(self.create_item(self.chosen_character)))
+                chosen_character_placed = True
+                continue
+            if item.name == "Crystal":
+                if not self.is_vanilla_game():
+                    (self.multiworld.get_location("Objective Reward", self.player)
+                     .place_locked_item(self.create_item("Crystal")))
+                    continue
+            if item.name.startswith("Objective"):
+                continue
+            self.multiworld.itempool.append(item)
+
+    def set_rules(self) -> None:
+        for location in locations.character_locations:
+            add_item_rule(self.multiworld.get_location(location, self.player),
+                     lambda item: item.name in items.characters and item.player == self.player)
+            if self.options.NoFreeCharacters.current_key == "true":
+                if location in locations.free_character_locations:
+                    add_item_rule(self.multiworld.get_location(location, self.player),
+                                  lambda item: item.name == "None")
+            if self.options.NoEarnedCharacters.current_key == "true":
+                if location in locations.earned_character_locations:
+                    add_item_rule(self.multiworld.get_location(location, self.player),
+                                  lambda item: item.name == "None")
+            if len(self.options.AllowedCharacters.value.difference(self.options.RestrictedCharacters.value)) > 0:
+                if location in locations.restricted_character_locations:
+                    add_item_rule(self.multiworld.get_location(location, self.player),
+                                  lambda item: item.name not in self.options.RestrictedCharacters.value)
+
+        for location in locations.all_locations:
+            if location.name not in locations.character_locations:
+                if location.name.startswith("Objective") or location.name == "Kokkol's House 2F -- Kokkol -- forge item":
+                    try:
+                        self.multiworld.get_location(location.name, self.player)
```

Comment:
```suggestion
                        self.get_location(location.name)
```
There's a `world.get_location`, which is just a quicker version of what you were doing. Feel free to swap to it or not